impeller/entity/shaders/filters/morphology_filter.frag

```diff
@@ -12,15 +12,15 @@ layout(constant_id = 0) const float supports_decal = 1.0;
 
 // These values must correspond to the order of the items in the
 // 'FilterContents::MorphType' enum class.
-const float16_t kMorphTypeDilate = 0.0hf;
-const float16_t kMorphTypeErode = 1.0hf;
+const float kMorphTypeDilate = 0.0;
```

Comment:
Even with these changes, the duplicate constant error is still showing up:

```
shaderc: internal error: compilation succeeded but failed to optimize: ID '27' decorated with RelaxedPrecision multiple times is not allowed.
  %float_0 = OpConstant %float 0
```

Also, similar changes are needed to: https://github.com/flutter/engine/blob/132d364ef37325afcc3fedaaae88b8cf1a1a6f92/impeller/entity/shaders/filters/yuv_to_rgb_filter.frag#L16-L17

Reply:
Updated. Sorry, do you mean that even with this change there are still compilation errors?

Reply:
Yeah that's what I meant. The compiler doesn't like 0 and 1 as constants for this file and `yuv_to_rgb_filter.frag`.

Reply:
The latest commit [80579dd](https://github.com/flutter/engine/pull/52213/commits/80579dd2fd5b155cec77eac8931401ca6bdea343) fixes the issue 🙂

Reply:
woot, thanks for helping with this @jiahaog